src/libraries/System.Private.CoreLib/src/System.Private.CoreLib.Shared.projitems

```diff
@@ -9,7 +9,22 @@
   <PropertyGroup>
     <Nullable>enable</Nullable>
     <IsOSXLike Condition="'$(TargetsOSX)' == 'true' or '$(TargetsiOS)' == 'true' or '$(TargetstvOS)' == 'true'">true</IsOSXLike>
+    <ILLinkSharedDirectory>$(MSBuildThisFileDirectory)ILLink\</ILLinkSharedDirectory>
   </PropertyGroup>
+  <ItemGroup>
+    <ILLinkSubstitutionsXmls Include="$(ILLinkSharedDirectory)ILLink.Substitutions.Shared.xml" />
+    <ILLinkSubstitutionsXmls Include="$(ILLinkSharedDirectory)ILLink.Substitutions.LittleEndian.xml" />
+
+    <ILLinkSubstitutionsXmls Include="$(ILLinkSharedDirectory)ILLink.Substitutions.64bit.xml"
+                             Condition="'$(Platform)' == 'x64' or '$(Platform)' == 'arm64'" />
+    <ILLinkSubstitutionsXmls Include="$(ILLinkSharedDirectory)ILLink.Substitutions.32bit.xml"
+                             Condition="'$(Platform)' == 'x86' or '$(Platform)' == 'arm' or '$(Platform)' == 'wasm'" />
+
+    <ILLinkSubstitutionsXmls Include="$(ILLinkSharedDirectory)ILLink.Substitutions.NoArmIntrinsics.xml"
+                             Condition="'$(Platform)' == 'x64' or '$(Platform)' == 'x86' or '$(Platform)' == 'wasm'" />
+    <ILLinkSubstitutionsXmls Include="$(ILLinkSharedDirectory)ILLink.Substitutions.NoX86Intrinsics.xml"
+                             Condition="'$(Platform)' == 'arm64' or '$(Platform)' == 'arm' or '$(Platform)' == 'wasm'" />
```

Comment:
Would it make sense to share the conditions with `*.PlatformNotSupported.cs` files below?

Reply:
I can. Do we still need the `$(FeatureHardwareIntrinsics)` flag? Both coreclr and mono set it to `true`, so I don't think it is necessary anymore.

Reply:
It should be fine to delete `FeatureHardwareIntrinsics` while you are on it

Reply:
It isn't true for ARM32

Reply:
(nor is FEATURE_SIMD, if that matters for `System.Numerics` intrinsics)

Reply:
`FeatureHardwareIntrinsics` is always true: https://github.com/dotnet/runtime/blob/master/src/coreclr/clr.featuredefines.props#L6

Reply:
What isn't true for ARM32? The places I see this property being set are:

https://github.com/dotnet/runtime/blob/f62e93416a1799aecc6b0947adad55a0d9870732/src/coreclr/clr.featuredefines.props#L6

and

https://github.com/dotnet/runtime/blob/a64f2ec1c837be82fbff4cadcdbfea94a4b100b9/src/mono/netcore/System.Private.CoreLib/System.Private.CoreLib.csproj#L132

Both projects set this property to true unconditionally.

Reply:
Ah, I see. `FeatureHwIntrinsics` isn't the same switch that controls `FEATURE_HW_INTRINSICS`.

The former controls the managed code while the latter controls the native code and they do not necessarily line up (the former is on for ARM32, while the latter is off, for example).